api/logical.go

```diff
@@ -24,6 +24,20 @@ func (c *Logical) Read(path string) (*Secret, error) {
 	return ParseSecret(resp.Body)
 }
 
+func (c *Logical) List(path string) (*Secret, error) {
+	r := c.c.NewRequest("LIST", "/v1/"+path)
+	resp, err := c.c.RawRequest(r)
+	if resp != nil && resp.StatusCode == 404 {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
```

Comment:
You are probably right.

I'll fix that in this branch, and fix that in the other functions in master.